app/src/main/java/woowacourse/shopping/feature/product/detail/dialog/CounterDialog.kt

```diff
@@ -0,0 +1,84 @@
+package woowacourse.shopping.feature.product.detail.dialog
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.core.os.bundleOf
+import androidx.fragment.app.DialogFragment
+import woowacourse.shopping.data.datasource.cartdatasource.CartLocalDataSourceImpl
+import woowacourse.shopping.databinding.FragmentCounterDialogBinding
+import woowacourse.shopping.feature.cart.CartActivity
+import woowacourse.shopping.feature.extension.getParcelableCompat
+import woowacourse.shopping.feature.list.item.ProductView.CartProductItem
+
+class CounterDialog : DialogFragment(), CounterDialogContract.View {
+    override lateinit var presenter: CounterDialogContract.Presenter
+    private var _binding: FragmentCounterDialogBinding? = null
+    private val binding: FragmentCounterDialogBinding
+        get() = _binding!!
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?,
+    ): View {
+        _binding = FragmentCounterDialogBinding.inflate(inflater, container, false)
+        return binding.root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        val product =
+            arguments?.getParcelableCompat<CartProductItem>(PRODUCT_KEY) ?: return dismiss()
+        val cartDb = CartLocalDataSourceImpl(requireContext())
+        presenter = CounterDialogPresenter(this, cartDb, product)
+        binding.product = product
+        setListener()
+
+        presenter.loadInitialData()
+    }
+
+    override fun setCountState(count: Int) {
+        binding.productCountTv.text = "$count"
+    }
+
+    private fun setListener() {
+        binding.addToCartButton.setOnClickListener {
+            presenter.addCart()
+            CartActivity.startActivity(requireContext())
+        }
+        binding.productPlusTv.setOnClickListener { presenter.updateCount(PLUS) }
+        binding.productMinusTv.setOnClickListener {
+            if ((binding.productCountTv.text == MIN_COUNT).not()) {
+                presenter.updateCount(MINUS)
+            }
+        }
+    }
+
+    override fun exit() {
+        dismiss()
+    }
+
+    override fun onDestroyView() {
+        super.onDestroyView()
+        _binding = null
+    }
+
+    companion object {
+        const val PRODUCT_KEY = "product"
+        const val COUNTER_DIALOG_TAG = "counter_dialog"
+        private const val MIN_COUNT = "1"
+        private const val PLUS = 1
+        private const val MINUS = -1
+
+        @JvmStatic
```

Comment:
```suggestion
        @JvmStatic
```
JvmStatic 어노테이션을 붙인 이유는 무엇일까요?

Reply:
프래그먼트를 생성하면 자동으로 생성되는 코드였기에 의미를 생각하지 않고 사용했던 것 같습니다!
제가 알고있는 `@JvmStatic`어노테이션은 클래스 파일을 컴파일 할 때 해당 메서드를 정적 메서드로 처리할 수 있도록 도와주는 역할을 합니다.

어노테이션을 제거하고 앱을 실행시켰을 때도 큰 문제 없이 동작하는 것 같았습니다. 
프래그먼트를 생성할 때 위 어노테이션이 붙는 이유에 대해 여쭤봐도 괜찮을까요?

Reply:
Empty Fragment를 만드신다면 어노테이션이 붙지 않을겁니다.
그리고 만약 붙는다하더라도 이게 정말로 필요한 것인지 의심을 해보셔야 합니다.
자동완성을 해준다고 무작정 쓰시는 것은 지양하시는 것이 좋습니다. 😄 

JvmStatic 어노테이션에 대해서 설명을 드리자면, 주로 Java와 Kotlin이 섞여있는 프로젝트에서 사용합니다.
Java에서는 Kotlin의 companion object를 접근하지 못하기에, 해당 어노테이션을 붙여줌으로써 Java에서도 접근하도록 합니다.
companion object는 마치 static으로 보이지만, 실제로는 static이 아닙니다.
companion object는 내부적으로 어떤 구조로 되어있는지 한번 찾아보시면 좋을 것 같아요.